crates/pixi_manifest/src/validation.rs

```diff
@@ -148,8 +149,39 @@ impl WorkspaceManifest {
             }
         }
 
-        // If there is a build section, make sure the build-string is not empty
+        // Check if the pixi build feature is enabled
+        let build_enabled = self
+            .workspace
+            .preview
+            .as_ref()
+            .map(|p| p.is_enabled(KnownPreviewFeature::PixiBuild))
+            .unwrap_or(false);
+
+        // Error any conda source dependencies are used and is not set
+        if !build_enabled {
+            let supported_platforms = self.workspace.platforms.as_ref();
+            // Check all features for source dependencies
+            for feature in self.features.values() {
+                if is_using_source_deps(feature, supported_platforms.iter()) {
+                    return Err(miette::miette!(
+                        help = "enable the `build` preview feature to use source dependencies",
+                        "source dependencies are used in the feature '{}', but the `pixi-build` preview feature is not enabled",
+                        feature.name
+                    ));
+                }
+            }
+        }
+
         if let Some(build) = &self.build {
+            // Check if we have enabled the build feature if we have a build section
+            if !build_enabled {
+                return Err(miette::miette!(
+                    help = "enable the `build` preview feature to use the build section",
```

Comment:
```suggestion
                    help = "enable the build preview feature to use the build section by setting `preview = ["pixi-build"]`",
```
Or something better ;)